src/main/java/seedu/address/model/person/Person.java

```diff
@@ -21,20 +21,39 @@
     private final Name name;
     private final Phone phone;
     private final Email email;
+    private Favourite favourite;
 
     // Data fields
     private final Address address;
     private final Optional<Property> property;
     private final Set<Tag> tags = new HashSet<>();
 
+    /**
+     * This constructor is used when editing a Client.
+     * Favourited clients will remain favourited & unfavourited clients will remain unfavourited
+     */
+    public Person(Name name, Phone phone, Email email, Favourite favourite, Address address,
+                  Optional<Property> property, Set<Tag> tags) {
+        requireAllNonNull(name, phone, email, favourite, address, property, tags);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.favourite = favourite;
+        this.property = property;
+        this.address = address;
+        this.tags.addAll(tags);
+    }
+
     /**
      * Every field must be present and not null.
+     * This constructor is used for adding a new Client, thus default status is unfavourited(false)
      */
     public Person(Name name, Phone phone, Email email, Address address, Optional<Property> property, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, property, tags);
+        requireAllNonNull(name, phone, email, address, tags);
```

Comment:
Is there any reason for removing `property` from `requireAllNonNull()`?

Reply:
Thanks for the confirmation, cause this was a newly created constructor that was written before your PR. Will include the `property` field in.